samples/scales/time/financial.html

```diff
@@ -53,11 +74,16 @@
 			var date = moment('Jan 01 1990', 'MMM DD YYYY');
 			var now = moment();
 			var data = [];
-			var unit = document.getElementById('unit').value;
-			for (; data.length < 60 && date.isBefore(now); date = date.clone().add(1, unit)) {
-				if (date.isoWeekday() <= 5) {
-					data.push(randomBar(date, data.length > 0 ? data[data.length - 1].y : 30));
+			for (; data.length < 60 && date.isBefore(now); date = date.clone().add(1, unit).startOf(unit)) {
+				if (outsideMarketHours(date)) {
+					if (unitLessThanDay() && !beforeNineThirty(date)) {
+						date = date.clone().add(date.isoWeekday() > 5 ? 8 - date.isoWeekday() : 1, 'day');
+					}
```

Comment:
The date needs to be incremented when the unit is 'day' or greater, OR it is after the market hours. `date.isoWeekday() >= 5` is important because it needs to be skipped to Monday if it is after the market hours on Friday.
```js
if (!unitLessThanDay() || !beforeNineThirty(date)) {
    date = date.clone().add(date.isoWeekday() >= 5 ? 8 - date.isoWeekday() : 1, 'day');
}
```

Reply:
`> 5` should be `>= 5`. Otherwise, you see data for Saturday 9:30am when the unit is 'hour'.
```
Array (60)
...
33 {t: "Fri Jan 05 1990 10:00:00 GMT+0800", y: "19.17"}
34 {t: "Fri Jan 05 1990 11:00:00 GMT+0800", y: "19.25"}
35 {t: "Fri Jan 05 1990 12:00:00 GMT+0800", y: "18.59"}
36 {t: "Fri Jan 05 1990 13:00:00 GMT+0800", y: "18.75"}
37 {t: "Fri Jan 05 1990 14:00:00 GMT+0800", y: "18.19"}
38 {t: "Fri Jan 05 1990 15:00:00 GMT+0800", y: "18.32"}
39 {t: "Fri Jan 05 1990 16:00:00 GMT+0800", y: "19.06"}
40 {t: "Sat Jan 06 1990 09:30:00 GMT+0800", y: "20.34"}
41 {t: "Mon Jan 08 1990 09:30:00 GMT+0800", y: "20.26"}
42 {t: "Mon Jan 08 1990 10:00:00 GMT+0800", y: "20.81"}
...
```

Reply:
Yes, good call. If we had tests for samples, this would be a good candidate :-)

Reply:
I've pushed an update